File: medical-app/src/app/domain/Comment.ts
/**
 * Created by yana on 23.03.18.
 */
export class Comment{

  private _id: number;
  private _createdDate: Date;
  private _text: string;

  constructor() {
  }

  constructor(createdDate: Date, text: string) {
    this._createdDate = createdDate;
    this._text = text;
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get createdDate(): Date {
    return this._createdDate;
  }

  set createdDate(value: Date) {
    this._createdDate = value;
  }

  get text(): string {
    return this._text;
  }

  set text(value: string) {
    this._text = value;
  }
}

File: src/main/java/com/nandy/medicalbase/domain/Comment.java
package com.nandy.medicalbase.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.util.Date;

/**
 * Created by yana on 22.03.18.
 */
@Entity
public class Comment {

    @Id
    @GeneratedValue
    private long id;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createDate;

    private String text;

    @JsonIgnore
    @ManyToOne
    private Patient patient;

    private Comment() { }

    public Comment(final Patient patient, final String text, final Date createDate) {
        this.patient = patient;
        this.text = text;
        this.createDate = createDate;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public Date getCreateDate() {
        return createDate;
    }

    public void setCreateDate(Date createDate) {
        this.createDate = createDate;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Patient getPatient() {
        return patient;
    }

    public void setPatient(Patient patient) {
        this.patient = patient;
    }
}
